### PR TITLE
chore: update CODEOWNERS with new team designation

### PR DESCRIPTION
**Pull request by booc0mtaco**
### Summary:

- update the codeowners file with the proper team name
- update the author in package.json with the same handle

### Test Plan:

- [x] Manually tested my changes, and here are the details:
  - use visual codeowners check file in GH
  - check email alias in google groups

### CODE REVIEW

**Review by jeremiah-clothier — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "@chanzuckerberg/eds",
   "version": "12.0.0",
   "description": "The React-powered design system library for Chan Zuckerberg Initiative education web applications",
-  "author": "CZI <edu-frontend-infra@chanzuckerberg.com>",
+  "author": "CZI <edu-dev-foundations@chanzuckerberg.com>",
```
Comment (jeremiah-clothier) [inline package.json:5]: Actually, I don't think this email exists yet

**Review by jeremiah-clothier — CHANGES_REQUESTED** [review]

**Comment by jeremiah-clothier** [comment]
> * check email alias in google groups

Where can you check this? I can't find the team (at least when sending email invites)

File: package.json (hunk 1 above)
Comment (booc0mtaco) [inline package.json:5]: Checked and this does not exist yet.

```suggestion
  "author": "CZI <edu-frontend-infra@chanzuckerberg.com>",
```

Comment (booc0mtaco) [reply package.json:5]: @jeremiah-clothier when logged in, I hit groups.google.com and searched for the team alias, but it wasn't defined. I'll revert using that suggestion above

**Pull request merged**